src/main/java/org/codehaus/mojo/license/utils/UrlRequester.java

```diff
@@ -82,80 +82,85 @@ public static boolean isStringUrl( String data )
     }
 
     /**
-     * this method will send a simple GET-request to the given destination and will return the result as a
-     * string
+     * Returns the content of the resource pointed by the given URL as a string.
      *
      * @param url the resource destination that is expected to contain pure text
      * @return the string representation of the resource at the given URL
+     * @throws IOException If an I/O error occurs when retrieve of the content URL
      */
-    public static String getFromUrl( String url ) throws MojoExecutionException
+    public static String getFromUrl( String url ) throws IOException
     {
+        return getFromUrl( url, "UTF-8" );
+    }
 
+    /**
+     * Returns the content of the resource pointed by the given URL as a string.
+     *
+     * @param url the resource destination that is expected to contain pure text
+     * @param encoding the resource content encoding
+     * @return the string representation of the resource at the given URL
+     * @throws IOException If an I/O error occurs when retrieve of the content URL
+     */
+    public static String getFromUrl( String url, String encoding ) throws IOException
+    {
         // by RFC url is composed by <protocol>:<schema-part>.
         // Here could not be used the URL parser because classpath does not have a registered Handler
         String protocol = StringUtils.substringBefore( url, ":" ).toLowerCase();
-        Charset utf8 = Charset.forName( "UTF-8" );
+        Charset charset = Charset.forName( encoding );
 
         String result = null;
-        try
+        if ( CLASSPATH_PROTOCOL.equals( protocol ) )
         {
-            if ( CLASSPATH_PROTOCOL.equals( protocol ) )
+            ClassLoader classLoader = UrlRequester.class.getClassLoader();
+
+            String resource = url.substring( CLASSPATH_PROTOCOL.length() + 1 );
+            URL resourceUrl = classLoader.getResource( resource );
+            if ( resourceUrl == null )
             {
-                ClassLoader classLoader = UrlRequester.class.getClassLoader();
-    
-                String resource = url.substring( CLASSPATH_PROTOCOL.length() + 1 );
-                URL resourceUrl = classLoader.getResource( resource );
-                if ( resourceUrl == null )
-                {
-                    resourceUrl = ClassLoader.getSystemResource( resource );
-                }
-                if ( resourceUrl != null )
-                {
-                    result = IOUtils.toString( resourceUrl, utf8 );
-                }
-                else
-                {
-                    throw new MojoExecutionException( "The resource " + resource
-                            + " was not found in the maven plugin classpath" );
-                }
+                resourceUrl = ClassLoader.getSystemResource( resource );
```

Comment:
The fallback to system classloader is redundant. `UrlRequester.class.getClassLoader()` gives you the plugin classloader whose `getResource()` already delegates to the parent class loader (and thus ultimately to the system classloader). See also https://maven.apache.org/guides/mini/guide-maven-classloading.html

Reply:
ok